aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -132,7 +148,12 @@ pub(crate) fn run_script_prologue(
             MoveValue::U8(chain_id.id()),
             MoveValue::vector_u8(txn_data.script_hash.clone()),
         ];
-        (&APTOS_TRANSACTION_VALIDATION.script_prologue_name, args)
+        if has_randomness_annotation {
+            args.push(MoveValue::Bool(true));
```

Comment:
Shall we add a feature flag, depending on which we switch to the new scripts entirely? Otherwise it's awkward because 1. we always pass a `true` to this parameter and 2. we need to keep logic of the two set of scripts in sync.

Reply:
do you mean:
```rust
let (func, args) = if some_flag_is_set {
    let args = vec![arg0, ..., argn, has_randomness_annotation];
    (new_name, args)
} else {
    let args = vec![arg0, ..., argn];
    (old_name, args)
}
```
?

Reply:
> keep logic of the two set of scripts in sync.

you mean those move functions, the original and the v2?

i thought it anyway has to be like that: the original one is either disabled, or becomes a wrapper that call v2.

Reply:
Yeah I figured you've reused code to the maximum in Move anyway. That's fine.